### PR TITLE
Generalized funs

### PR DESCRIPTION
**Pull request by john-b-edwards**
generalized functions for scraping ncaa rosters and schedules

### CODE REVIEW

File: R/ncaa_schedule.R [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+      rvest::html_node('#contentarea') |>
+      rvest::html_node('#stats_player_person_id') |>
+      rvest::html_node('option') |>
+      rvest::html_text()
+    game_id_table <- data.frame(
+      Date = html_b |>
+        rvest::html_node('body') |>
+        rvest::html_node('div#contentarea') |>
+        rvest::html_node('div#game_breakdown_div') |>
+        rvest::html_nodes(xpath='table/tr/td/table/tr/td[count(table/tr/td/table/tr/th[.="Result"]/preceding-sibling::th)+1]') |>
```
Comment (akeaswaran) [inline R/ncaa_schedule.R:48]: jesus, this is an ugly one

File: R/ncaa_schedule.R [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+        rvest::html_node('div#game_breakdown_div') |>
+        rvest::html_nodes(xpath='table/tr/td/table/tr/td[count(table/tr/td/table/tr/th[.="Result"]/preceding-sibling::th)+2]') |>
+        lapply(\(x) {
+          if(length(rvest::html_attr(x,'class'))){
+            if(!is.na(rvest::html_attr(x,'class'))){
+              return(rvest::html_text(x,trim=T))
+            }
+          }}) |>
+        unlist() |>
+        stringi::stri_replace_all(regex = '((?=(.*)) @ (.*)$)|(@ )', ''),
```
Comment (akeaswaran) [inline R/ncaa_schedule.R:65]: How are we handling cancelled/postponed games here?

**Review by akeaswaran — APPROVED** [review]

**Review by akeaswaran — APPROVED** [review]

File: R/ncaa_schedule.R (hunk 1 above)
Comment (john-b-edwards) [reply R/ncaa_schedule.R:48]: xpath usually is lmao

the impetus behind this that we want to grab only tds that are in a given column for Dates. this was _about_ the best I could come up with.

File: R/ncaa_schedule.R (hunk 2 above)
Comment (john-b-edwards) [reply R/ncaa_schedule.R:65]: Believe that the left join in line 205 _does not_ filter cancelled/postponed games, you just scores/results listed as NA? should probably be an inner_join instead if we want to filter them.

File: R/ncaa_schedule.R [hunk 3]
```diff
@@ -136,8 +201,7 @@ ncaa_schedule <- function(team_id) {
         )
       )),
       T ~ NA_character_),
-      Opponent_Clean = Opponent |> stringi::stri_replace_all(regex = '^#([0-9]*)|(@ )|(@(.*)$)', '') |>
-        trimws()) |>
+      Opponent_Clean = trimws(gsub('@ |^\\#[0-9]* |^[0-9]* | x_break|^\\@#[0-9]* ','',stringi::stri_extract_all_regex(paste0(Opponent,' x_break'),'^.+?(?=x_break)')))) |>
     dplyr::left_join(opponent_id_table, by=c('Opponent_Clean')) |>
```
Comment (john-b-edwards) [inline R/ncaa_schedule.R:212]: should probably be an inner join if we want to drop cancelled/postponed games

**Review by john-b-edwards — COMMENTED** [review]
only part is swapping the inner to the left join that i can see

**Pull request merged**